### PR TITLE
method to determine default script for a section

### PR DESCRIPTION
**Pull request by Bjvanminnen**
It's now possible for a section to have a course assigned to it, but not have a script assigned to it (though no way to do this via UI yet). This PR exposes a method that determines the default script for a section.

This is needed, because when we navigate to a page like http://localhost.code.org:3000/teacher-dashboard#/sections/11/responses we ask for the text_responses for the current section (without providing a specific script_id). In the case where the section has a course, but not script, we want to show the responses for the first script in the course.

I had originally intended to also do a bunch of clean up in this PR. Almost all of the places where we call `load_script`/`load_section` to instantiate `@script`/`@section`, we do not actually need to be using instance variables. Unfortunately, our tests our heavily dependent on checking for the presence/value of these instance variables via `assigns`, so I left them alone for now.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
I think I might be able to replace the tests that use `assigns` with ones that just call load_script directly (my new version that returns a script instead of setting an instance variable). I'm going to explore that in a follow on PR.

File: dashboard/app/models/section.rb [hunk 1]
```diff
@@ -129,6 +129,14 @@ def add_and_remove_student(student, old_section)
     add_student student
   end
 
+  # Figures out the default script for this section. If the section is assigned to
+  # a course rather than a script, it returns the first script in that course
+  # @return [Script]
```
Comment (ashercodeorg) [inline dashboard/app/models/section.rb]: This can return nil, right?

File: dashboard/app/models/section.rb [hunk 2]
```diff
@@ -129,6 +129,14 @@ def add_and_remove_student(student, old_section)
...
+  def default_script
+    return script if script
+    return course.try(:course_scripts).try(:first).try(:script)
```
Comment (ashercodeorg) [inline dashboard/app/models/section.rb]: Is this meant to return based on script position (in the course) or DB order (as implemented)?

Comment (aoby) [reply dashboard/app/models/section.rb]: It is [position order](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/course.rb#L19), no?

Comment (ashercodeorg) [reply dashboard/app/models/section.rb]: Ah, my bad. Nice. :)

File: dashboard/app/models/section.rb (hunk 1 above)
Comment (Bjvanminnen) [reply dashboard/app/models/section.rb]: It can. What's the right way of indicating that?

Comment (aoby) [reply dashboard/app/models/section.rb]: `@return [Script, nil]`

**Pull request merged**